test/transaction/input/input.js

```diff
@@ -41,6 +49,39 @@ describe('Transaction.Input', function() {
   });
   it('detects coinbase transactions', function() {
     new Input(output).isNull().should.equal(false);
-    new Input(coinbase).isNull().should.equal(true);
+    var ci = new Input(coinbase);
+    ci.isNull().should.equal(true);
+  });
+
+  describe('instantiation', function() {
+    it('works without new', function() {
+      var input = Input();
+      should.exist(input);
+    });
+    it('fails with no script info', function() {
+      expect(function() {
+        var input = new Input({});
+        input.toString();
+      }).to.throw('Need a script to create an input');
+    });
+    it('fromJSON', function() {
+      var input = Input.fromJSON(coinbaseJSON);
+      var otherInput = Input.fromJSON(otherJSON);
+      should.exist(input);
+      should.exist(otherInput);
+    });
+    it('fromObject', function() {
+      var input = Input.fromJSON(coinbaseJSON);
+      var obj = input.toObject();
+      obj.script = new Buffer(obj.script, 'hex');
+      Input.fromObject(obj).should.deep.equal(input);
+      obj.script = 42;
+      Input.fromObject.bind(null, obj).should.throw('Invalid argument type: script');
+    });
+  });
+
+  it('_estimateSize', function() {
```

Comment:
Shouldn't this fail with a `NonImplementedError` or something?

Same as https://github.com/bitpay/bitcore/pull/1124/files#r25592034, please use a more descriptive name.

If this were the test for `PublicKeyHashInput`, I'd use something like `_estimateSize returns the maximum size a scriptSig for a P2PKH`